Shared/Samples/Show device location history/ShowDeviceLocationHistoryView.swift

```diff
@@ -0,0 +1,234 @@
+                } catch {
+                    // Shows an alert with an error if starting the data source fails.
+                    self.error = error
+                }
+            }
+            .onDisappear {
+                model.stopProcessingLocationChanges()
+                model.isTracking = false
+                model.stopLocationDataSource()
+            }
+            .toolbar {
+                ToolbarItem(placement: .bottomBar) {
+                    Button(model.isTracking ? "Stop Tracking" : "Start Tracking") {
+                        model.isTracking.toggle()
+                    }
+                    .disabled(trackingButtonIsDisabled)
+                }
+            }
+            .alert(isPresented: $isShowingAlert, presentingError: error)
+    }
+}
+
+private extension ShowDeviceLocationHistoryView {
+    /// The model used to store the geo model and other expensive objects
+    /// used in this view.
+    @MainActor
+    class Model: ObservableObject {
+        /// A Boolean value indicating whether to track the device location.
+        @Published var isTracking = false {
+            didSet {
+                if isTracking {
+                    startProcessingLocationChanges()
+                } else {
+                    stopProcessingLocationChanges()
+                }
+            }
+        }
+        
+        /// A map with a light gray basemap style.
+        let map = Map(basemapStyle: .arcGISLightGrayBase)
+        
+        /// A location display using a simulated location data source.
+        let locationDisplay: LocationDisplay
+        
+        /// A graphics overlay for location history polyline.
+        let trackOverlay: GraphicsOverlay
+        
+        /// A graphics overlay for location history points.
+        let locationsOverlay: GraphicsOverlay
+        
+        /// The markers for traveled location points.
+        private let locationGraphic = Graphic()
+        
+        /// The line graphic for traveled path.
+        private let trackGraphic = Graphic()
+        
+        /// A polyline builder to make the line graphic.
+        private let trackBuilder = PolylineBuilder(spatialReference: .webMercator)
+        
+        /// A multipoint builder to make the point graphics.
+        private let pointBuilder = MultipointBuilder(spatialReference: .webMercator)
+        
+        /// A task to get location updates from the asynchronous stream.
+        private var locationProcessingTask: Task<(), Never>?
+        
+        init() {
+            locationDisplay = LocationDisplay(dataSource: Self.makeSimulatedLocationDataSource())
+            locationDisplay.autoPanMode = .recenter
+            
+            trackOverlay = GraphicsOverlay(graphics: [trackGraphic])
+            trackOverlay.renderer = SimpleRenderer(
+                symbol: SimpleLineSymbol(
+                    style: .solid,
+                    color: UIColor(Color.accentColor.opacity(0.5)),
```

Comment:
This makes me think.. did we ever discuss swapping out `UIColor` with `Color` in the SDK? I imagine we did but stayed with `UIColor` for a reason?

Reply:
There was a discussion in `swift/issues/1491`. I think the concluded rule of thumb is to use `Color` when it's working with a SwiftUI `View` (like setting a color via view modifier), and use `UIColor` for other stuff that's displayed within the map/scene view.

Reply:
Hmm, interesting. Thanks!

Reply:
```suggestion
                    color: .tintColor.withAlphaComponent(0.5),
```